src/SPC/command/DownloadCommand.php

```diff
@@ -249,7 +235,24 @@ public function handle(): int
                         logger()->warning("Pre-built content not found for {$source}, fallback to source download");
                     }
                     logger()->info("[{$ni}/{$cnt}] Downloading source {$source}");
-                    Downloader::downloadSource($source, $config, $force_all || in_array($source, $force_list));
+                    try {
+                        Downloader::downloadSource($source, $config, $force_all || in_array($source, $force_list));
+                    } /* @noinspection PhpRedundantCatchClauseInspection */ catch (DownloaderException|RuntimeException $e) {
+                        // if `--no-alt` option is set, we will not download alternative sources
+                        if ($this->getOption('no-alt')) {
+                            throw $e;
+                        }
+                        // if download failed, we will try to download alternative sources
+                        logger()->warning("Download failed: {$e->getMessage()}");
+                        logger()->notice("Trying to download alternative sources for {$source}");
+                        $alt_sources = Config::getSource($source);
+                        if (!isset($alt_sources['alt'])) {
+                            $alt_config = array_merge($config, $this->getDefaultAlternativeSource($source));
+                        } else {
+                            $alt_config = array_merge($config, $alt_sources['alt']);
+                        }
```

Comment:
I would call the variable $sourceConfig or something, $alt_sources['alt'] is very confusing.
or `$alt_sources = Config::getSource($source)['alt'] ?? null`